### PR TITLE
completes task 1 - create GET /api/topics endpoint. Tests created to …

### PR DESCRIPTION
**Pull request by mikef80**
…check for response from API and API functionality broken down using MVC principles. No sad path created yet as no user definable input available to break endpoint

### CODE REVIEW

**Comment by mikef80** [comment]
To be clearer, this is for Step 2 of the project, not Step 1.  Sorry!

File: api/app.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const express = require('express');
+const { getTopics } = require('./controllers/topics-controllers');
+
+const app = express();
+
+app.use(express.json());
```
Comment (nc-mick-fay) [inline api/app.js:6]: We aren't posting anything just yet, so this line isn't actually do anything!

File: __tests__/topics.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  it("GET: 200 return array is populated with objects that have the required properties - 'slug' and 'description'", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        body.topics.forEach((topic) => {
+          expect(topic).toHaveProperty("slug");
+          expect(topic).toHaveProperty("description");
+        });
```
Comment (nc-mick-fay) [inline __tests__/topics.test.js:33]: Are you sure you're getting an array of all the topics back from your response? We know it's an array and that each item has those properties, but is it the right length?

**Review by nc-mick-fay — CHANGES_REQUESTED** [review]
Hi Mike, really nice endpoint and file structure so far!
I'd just like to see a test for the length of the array so we can be sure we're getting back all our topics.

A bonus thing to consider - how would you handle errors for bad endpoints? Don't have to add this in as express will handle these for us, but if you wanted to customise your responses it's something you might want to think about :)

File: __tests__/topics.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      });
+  });
+
+  it("GET:200 returns an array of the correct length", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.topics).toHaveLength(3);
```
Comment (Hard-Luck) [inline __tests__/topics.test.js:42]: I would probably just add this check to the previous test block before the forEach as it is being used to validate the tests in the forEach

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Optional but recommended change then you're good to merge

File: __tests__/topics.test.js (hunk 3 above)
Comment (mikef80) [reply __tests__/topics.test.js:42]: Cool, thank you.

**Pull request merged**